pkg/sfu/downtrack.go

```diff
@@ -952,6 +954,10 @@ func (d *DownTrack) retransmitPackets(nackedPackets []packetMeta) {
 	defer PacketFactory.Put(src)
 
 	for _, meta := range nackedPackets {
+		if !d.forwarder.IsRtxAllowed(int32(meta.layer)) {
```

Comment:
Limiting RTX. With NLC throttling, the bandwidth estimate dropped. But, just RTX alone on the 720p layer was reaching in excess of 3 Mbps some times and that added to further congestion and delays in getting a key frame for the lower layer (because PLI would have been queued/lost). Comments on when it is restricted in `Forwarder` module.

Reply:
For these packet should have same layer, can we only check IsRtxAllowd once for all nackedPackets?

Reply:
I thought about that. But, it is tricky with browsers sending NACKs. Even after we send a key frame after a layer switch, NACKs from previous layer keep coming in. It depends on the jitter buffer on client side. It keeps trying to get those packets till the key frame gets to the head of the queue and probably flushes the NACK queue.

BTW, we should also implement flush pending NACKs on getting a key frame packet. Another thing in my head TODO :-) I need to make tasks for these.

Reply:
I do see a different option though. We can make this API return list of allowed RTX layers and the calling code can filter out disallowed layers. I will make that change in a subsequent PR.